test/components/braveryPanelTest.js

```diff
@@ -268,6 +269,56 @@ describe('Bravery Panel', function () {
         .click(blockAdsOption)
         .waitForTextValue(adsBlockedStat, '2')
     })
+    it('blocks websocket tracking', function * () {
+      const url = Brave.server.url('websockets.html')
+      let time1 = 0
+      let time2 = 0
+      yield this.app.client
+        .waitForDataFile('adblock')
+        .tabByIndex(0)
+        .loadUrl(url)
+        .waitUntil(function () {
+          return this.getText('#result')
+            .then((result) => {
+              return result === 'success'
+            })
+        })
+        .waitUntil(function () {
+          // The connection to the websocket URL is blocked by adblock, so
+          // onerror fires quickly
+          return this.getText('#errorTime')
+            .then((time) => {
+              time = Number(time)
+              if (time > 0) {
+                time1 = time
+                return true
+              }
+              return false
+            })
+        })
+        // TODO: Show websocket URLs in Bravery Stats
+        .openBraveMenu(braveMenu, braveryPanel)
+        .click(adsBlockedControl)
+        .waitForVisible(showAdsOption)
+        .click(showAdsOption)
+        .tabByIndex(0)
+        .waitUntil(function () {
+          // The connection to the websocket URL is allowed by adblock, so
+          // onerror fires only after the connection is attempted
+          return this.getText('#errorTime')
+            .then((time) => {
+              time = Number(time)
+              if (time > 0) {
+                time2 = time
+                return true
+              }
+              return false
+            })
+        })
+      // Assume blocking websockets is working if the time difference between
+      // onerror firing with adblock on vs off is sufficiently high.
+      assert(time2 - time1 > 100)
```

Comment:
@bridiver no because "ws://ag.innovid.com/dv/sync?tid=2" doesn't support websockets. it fails either way: either because it's blocked by adblock (the fast case) or because the server response is incorrect (the slow case)